src/index.js

```diff
@@ -353,12 +360,31 @@ module.exports = (
             har.postData.mimeType = 'multipart/form-data';
             har.postData.params = [];
 
-            // Discover all `{ type: string, format: binary }` properties the schema. If there are
-            // any, then that means that we're dealing with a `multipart/form-data` request and
-            // need to treat the payload as `postData.params`.
-            const binaryTypes = Object.keys(requestBody.schema.properties).filter(
-              key => requestBody.schema.properties[key].format === 'binary'
-            );
+            /**
+             * Discover all `{ type: string, format: binary }` properties, or arrays containing the
+             * same, within the request body. If there are any, then that means that we're dealing
+             * with a `multipart/form-data` request and need to treat the payload as
+             * `postData.params` and supply filenames and content types for the files (if they're
+             * available).
+             *
+             * @example `{ type: string, format: binary }`
+             * @example `{ type: array, items: { type: string, format: binary } }`
+             */
+            const binaryTypes = Object.keys(requestBody.schema.properties).filter(key => {
+              if (requestBody.schema.properties[key].format === 'binary') {
+                return true;
+              } else if (
+                requestBody.schema.properties[key].type === 'array' &&
+                requestBody.schema.properties[key].items &&
+                typeof requestBody.schema.properties[key].items === 'object' &&
+                requestBody.schema.properties[key].items !== null &&
+                requestBody.schema.properties[key].items.format === 'binary'
```

Comment:
This work is annoying but it's valid JSON Schema for `items` to be `{}` or `false`.